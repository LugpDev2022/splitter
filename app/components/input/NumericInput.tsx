interface NumericInputProps {
  hasError?: boolean;
  onChange: (target: any) => void;
  value: number;
}

const NumericInput: React.FC<NumericInputProps> = ({
  hasError,
  onChange,
  value,
}) => (
  <input
    type='text'
    className={`
  ${hasError ? 'border-red-500' : 'border-transparent'}
  bg-[#f4fafa]
  border-2
  duration-300
  focus:border-cyan-400
  font-bold
  hover:border-cyan-400
  hover:cursor-pointer 
  outline-none
  p-2
  rounded-lg
  text-[#00494d]
  text-2xl
  text-right
  transition
  w-full
  `}
    placeholder='0'
    value={value > 0 ? value.toString() : ''}
    onChange={onChange}
  />
);

export default NumericInput;
