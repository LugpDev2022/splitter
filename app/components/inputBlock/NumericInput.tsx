'use client';

import { InputType } from './types';
import { useNumericInput } from './useNumericInput';

interface NumericInputProps {
  hasError?: boolean;
  inputType: InputType;
}

const NumericInput: React.FC<NumericInputProps> = ({ hasError, inputType }) => {
  const { onInputChange, onBlur, value } = useNumericInput(inputType);

  return (
    <input
      type='text'
      className={`
      ${hasError ? 'border-red-500' : 'border-transparent'}
      bg-[#f4fafa]
      border-2
      duration-300
      focus:border-cyan-400
      font-bold
      hover:border-cyan-400
      hover:cursor-pointer 
      outline-none
      p-2
      rounded-lg
      text-[#00494d]
      text-2xl
      text-right
      transition
      w-full
      `}
      placeholder='0'
      value={value || value === 0 ? value : ''}
      onChange={onInputChange}
      onBlur={onBlur}
    />
  );
};

export default NumericInput;
