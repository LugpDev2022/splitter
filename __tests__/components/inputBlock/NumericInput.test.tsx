import NumericInput from '@/app/components/inputBlock/NumericInput';
import { fireEvent, render, screen } from '@testing-library/react';

describe('tests on <NumericInput />', () => {
  const onChange = jest.fn();
  const onBlur = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(
      <NumericInput onBlur={onBlur} onChange={onChange} value={0} />
    );

    expect(container).toMatchSnapshot();
  });

  test('should have a border when has error', () => {
    render(
      <NumericInput onBlur={onBlur} onChange={onChange} hasError value={0} />
    );

    const input = screen.getByRole('textbox');
    expect(input.getAttribute('class')).toContain('border-red-500');
    expect(input.getAttribute('class')).not.toContain('border-transparent');
  });

  test('should not have a border when has error', () => {
    render(<NumericInput onBlur={onBlur} onChange={onChange} value={0} />);

    const input = screen.getByRole('textbox');
    expect(input.getAttribute('class')).not.toContain('border-red-500');
    expect(input.getAttribute('class')).toContain('border-transparent');
  });

  test('should call the onChange function', () => {
    render(<NumericInput onBlur={onBlur} onChange={onChange} value={0} />);

    const input = screen.getByRole('textbox');

    fireEvent.change(input, { target: { value: '7' } });
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  test('should call the onBlur function', () => {
    render(<NumericInput onBlur={onBlur} onChange={onChange} value={0} />);

    const input = screen.getByRole('textbox');

    fireEvent.blur(input);
    expect(onBlur).toHaveBeenCalledTimes(1);
  });
});
