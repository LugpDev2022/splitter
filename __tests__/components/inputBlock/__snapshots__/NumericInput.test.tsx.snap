// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <NumericInput /> should match snapshot 1`] = `
<div>
  <input
    class="
    border-transparent
    bg-[#f4fafa]
    border-2
    duration-300
    focus:border-cyan-400
    font-bold
    hover:border-cyan-400
    hover:cursor-pointer 
    outline-none
    p-2
    rounded-lg
    text-[#00494d]
    text-2xl
    text-right
    transition
    w-full
    "
    placeholder="0"
    type="text"
    value="0"
  />
</div>
`;
